pkgs/by-name/ma/matrix-continuwuity/package.nix

```
@@ -13,35 +13,94 @@
   enableBlurhashing ? true,
   # upstream continuwuity enables jemalloc by default, so we follow suit
   enableJemalloc ? true,
-  rust-jemalloc-sys,
+  rust-jemalloc-sys-unprefixed,
   enableLiburing ? stdenv.hostPlatform.isLinux,
   liburing,
   nixosTests,
 }:
 let
-  rust-jemalloc-sys' = rust-jemalloc-sys.override {
-    unprefixed = !stdenv.hostPlatform.isDarwin;
-  };
-  rocksdb' = rocksdb.override {
-    inherit enableLiburing;
-    # rocksdb does not support prefixed jemalloc, which is required on darwin
-    enableJemalloc = enableJemalloc && !stdenv.hostPlatform.isDarwin;
-    jemalloc = rust-jemalloc-sys';
-  };
+  rocksdb' =
+    (rocksdb.override {
+      inherit enableLiburing;
+      # rocksdb does not support prefixed jemalloc, which is required on darwin
+      enableJemalloc = enableJemalloc && !stdenv.hostPlatform.isDarwin;
+      jemalloc = rust-jemalloc-sys-unprefixed;
+    }).overrideAttrs
+      (
+        final: old: {
+          version = "10.4.fb";
+          src = fetchFromGitea {
+            domain = "forgejo.ellis.link";
+            owner = "continuwuation";
+            repo = "rocksdb";
+            rev = "10.4.fb";
+            hash = "sha256-/Hvy1yTH/0D5aa7bc+/uqFugCQq4InTdwlRw88vA5IY=";
+          };
+
+          patches = [ ];
+
+          cmakeFlags =
+            lib.subtractLists [
+              # no real reason to have snappy or zlib, no one uses this
+              "-DWITH_SNAPPY=1"
```

```suggestion
              (lib.cmakeBool "WITH_SNAPPY" true)
```
Nit (non-blocking) prefer `lib.cmakeBool for these.